src/bin/ion/commands/generate/templates/java/enum.templ

```diff
@@ -0,0 +1,63 @@
+{% set full_namespace = namespace | join(sep=".") %}
+{% if is_nested == false %}
+package {{ full_namespace }};
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonException;
+import com.amazon.ion.IonWriter;
+import com.amazon.ion.IonType;
+import java.io.IOException;
+{% endif %}
+
+{# Verify that the abstract data type is a enum and store information for this enum #}
+{% set enum_info = model.code_gen_type["Enum"] %}
+
+{% if is_nested == true %} static {% endif %} enum {{ model.name }} {
+    {% for variant in enum_info["variants"] -%}
+           {{ variant.name | upper }} {
```

Comment:
Does tera have an `upper_snake` filter? It would be more useful in a case like this because then you won't have a name collision for `valid_values: [FooDataBar, FoodAtABar]`.

Reply:
Using filter `snake` then `upper`